modules/xmpp/JingleSessionPC.js

```diff
@@ -922,6 +922,26 @@ export default class JingleSessionPC extends JingleSession {
             const workFunction = (finishedCallback) => {
                 const oldSdp
                     = new SDP(this.peerconnection.localDescription.sdp);
+                // NOTE the code below assumes that no more than 1 video track
+                // can be added to the peer connection.
+                // Transition from no video to video (possibly screen sharing)
+                if (!oldTrack && newTrack && newTrack.isVideoTrack()) {
+                    // Clearing current primary SSRC will make
+                    // the SdpConsistency generate a new one which will result
+                    // with:
+                    // 1. source-remove for the recvonly
+                    // 2. source-add for the new video stream
+                    this.peerconnection.clearRecvonlySsrc();
+                }
+                // Transition from video to no video
+                if (oldTrack && oldTrack.isVideoTrack() && !newTrack) {
```

Comment:
not sure what kind of clarity this would add... how I personally see it that it would only make me process also the previous statement in combination with this one. Do we really have to be so picky ?